src/psyclone/transformations.py

```diff
@@ -1021,56 +1230,73 @@ class to convert the Loop represented by :py:obj:`node` into a
         return schedule, keep
 
 
-class OMPParallelTrans(Transformation):
+@six.add_metaclass(abc.ABCMeta)
+class ParallelRegionTrans(Transformation):
 
-    ''' Create an OpenMP PARALLEL region by inserting directives. For
-    example:
-
-    >>> from psyclone.parse import parse, ParseError
-    >>> from psyclone.psyGen import PSyFactory, GenerationError
-    >>> api="gocean1.0"
-    >>> filename="nemolite2d_alg.f90"
-    >>> ast,invokeInfo=parse(filename,api=api,invoke_name="invoke")
-    >>> psy=PSyFactory(api).create(invokeInfo)
-    >>>
-    >>> from psyclone.psyGen import TransInfo
-    >>> t=TransInfo()
-    >>> ltrans = t.get_trans_name('GOceanOMPLoopTrans')
-    >>> rtrans = t.get_trans_name('OMPParallelTrans')
-    >>>
-    >>> schedule=psy.invokes.get('invoke_0').schedule
-    >>> schedule.view()
-    >>> new_schedule=schedule
-    >>>
-    >>> # Apply the OpenMP Loop transformation to *every* loop
-    >>> # in the schedule
-    >>> for child in schedule.children:
-    >>>     newschedule,memento=ltrans.apply(child)
-    >>>     schedule = newschedule
-    >>>
-    >>> # Enclose all of these loops within a single OpenMP
-    >>> # PARALLEL region
-    >>> newschedule, _ = rtrans.apply(schedule.children)
-    >>> newschedule.view()
+    '''
+    Base class for transformations that create a parallel region.
 
     '''
+    def __init__(self):
+        self._pdirective = None  # Holds the class instance for the
+                                 # type of parallel region to generate
+        Transformation.__init__(self)
 
+    @abc.abstractmethod
     def __str__(self):
-        return "Insert an OpenMP Parallel region"
+        pass  # pragma: no cover
 
-    @property
+    @abc.abstractproperty
     def name(self):
         ''' Returns the name of this transformation as a string.'''
-        return "OMPParallelTrans"
```

Comment:
Add return type information as is done for the parallellooptrans abstract class

Reply:
Done.